### PR TITLE
Subsystem Initialization Checks

### PR DESCRIPTION
**Pull request by eandr127**
## Summary of Changes
- Log initialization results for all subsystems
- Check talons, encoders, gyro for drivebase
- Disable closed loop modes for drivebase if errors occur
- Check lift talon, encoder, and limit switch in initialization
- Disable closed loop motion for lift if errors occur
- Check intake Victor in initialization

## Testing Performed
**Environment**:  Practice bot

### CODE REVIEW

**Comment by kevlam2706** [comment]
Looks great. We still should test the lift encoder failure case by disconnecting the encoder cable from talon #5. Couldn't do it tonight because we didn't have allen keys at Mission Control.

Only suggestion I have, is you could log the error value that you get when it's not ErrorCode.OK. com.ctre.phoenix.ErrorCode has a number of possible error values that might be useful for debugging.

File: src/main/java/ca/team2706/frc/robot/subsystems/DriveBase.java [hunk 1]
```diff
@@ -454,6 +516,11 @@ public void setOpenLoopVoltageMode() {
      * Sets the talons to a disabled mode
      */
     public void setPositionNoGyroMode() {
+        if(!canRunAuto()) {
+            setDisabledMode();
+            return;
```
Comment (KyleRAnderson) [inline src/main/java/ca/team2706/frc/robot/subsystems/DriveBase.java]: I'm not sure I like the early return here. I would prefer an `if...else` structure.

File: src/main/java/ca/team2706/frc/robot/subsystems/DriveBase.java [hunk 2]
```diff
@@ -467,6 +534,11 @@ public void setPositionNoGyroMode() {
      * Sets the robot up for rotation.
      */
     public void setRotateMode() {
+        if(!canRunAuto()) {
+            setDisabledMode();
+            return;
```
Comment (KyleRAnderson) [inline src/main/java/ca/team2706/frc/robot/subsystems/DriveBase.java]: Same thing with early return as mentioned before.

File: src/main/java/ca/team2706/frc/robot/subsystems/DriveBase.java [hunk 3]
```diff
@@ -480,6 +552,11 @@ public void setRotateMode() {
      * Sets motion magic
      */
     public void setMotionMagicWithGyroMode() {
+        if(!canRunAuto()) {
+            setDisabledMode();
+            return;
```
Comment (KyleRAnderson) [inline src/main/java/ca/team2706/frc/robot/subsystems/DriveBase.java]: Same issue with early return.

File: src/main/java/ca/team2706/frc/robot/subsystems/DriveBase.java [hunk 4]
```diff
@@ -494,6 +571,11 @@ public void setMotionMagicWithGyroMode() {
      * Sets the drive mode to motion profile
      */
     public void setMotionProfile() {
+        if(!canRunAuto()) {
+            setDisabledMode();
+            return;
```
Comment (KyleRAnderson) [inline src/main/java/ca/team2706/frc/robot/subsystems/DriveBase.java]: Early return.

File: src/main/java/ca/team2706/frc/robot/subsystems/DriveBase.java [hunk 5]
```diff
@@ -509,6 +591,11 @@ public void setMotionProfile() {
      * Sets the drive mode to 2 wheel motion profile
      */
     public void setMotionProfile2Wheel() {
+        if(!canRunAuto()) {
+            setDisabledMode();
+            return;
```
Comment (KyleRAnderson) [inline src/main/java/ca/team2706/frc/robot/subsystems/DriveBase.java]: Early return.

File: src/main/java/ca/team2706/frc/robot/subsystems/DriveBase.java [hunk 6]
```diff
@@ -547,6 +634,11 @@ private void configMotionProfile() {
      * Gets the encoder sum using the pigeon
      */
     public void setPositionGyroMode() {
+        if(!canRunAuto()) {
+            setDisabledMode();
+            return;
```
Comment (KyleRAnderson) [inline src/main/java/ca/team2706/frc/robot/subsystems/DriveBase.java]: Early return.

File: src/main/java/ca/team2706/frc/robot/subsystems/Intake.java [hunk 7]
```diff
@@ -59,6 +75,9 @@ private Intake() {
                 new AnalogInput(Config.CARGO_IR_SENSOR_ID));
     }
 
+    public SubsystemStatus getStatus() {
```
Comment (KyleRAnderson) [inline src/main/java/ca/team2706/frc/robot/subsystems/Intake.java]: Javadoc.

File: src/main/java/ca/team2706/frc/robot/subsystems/Lift.java [hunk 8]
```diff
@@ -67,7 +73,11 @@ private Lift() {
         liftMotor = new WPI_TalonSRX(Config.LIFT_MOTOR_ID);
         addChild("Lift Motor", liftMotor);
         addChild("Lift Position", Sendables.newTalonEncoderSendable(liftMotor));
-        setupTalonConfig();
+        status = setupTalonConfig();
+    }
+
+    private SubsystemStatus getStatus() {
```
Comment (KyleRAnderson) [inline src/main/java/ca/team2706/frc/robot/subsystems/Lift.java]: Javadoc.

File: src/main/java/ca/team2706/frc/robot/subsystems/Lift.java [hunk 9]
```diff
@@ -236,6 +265,10 @@ public void setPositionMotionMagicEncoderTicks(final double maxSpeed, final doub
      * @param velocity Velocity in encoder ticks.
      */
     public void setVelocity(int velocity) {
+        if(getStatus() == SubsystemStatus.DISABLE_AUTO || getStatus() == SubsystemStatus.ERROR) {
+            return;
```
Comment (KyleRAnderson) [inline src/main/java/ca/team2706/frc/robot/subsystems/Lift.java]: Early return.

**Pull request merged**